internal/armada/server/submit.go

```diff
@@ -270,10 +271,15 @@ func (server *SubmitServer) SubmitJobs(ctx context.Context, req *api.JobSubmitRe
 
 	q, err := server.getQueueOrCreate(ctx, req.Queue)
 	if err != nil {
-		return nil, err
+		code := codes.Unknown
+		if e, ok := status.FromError(err); ok {
+			code = e.Code()
```

Comment:
Maybe one for a separate PR but we now have something that magically sets the return code base on error type:
https://github.com/G-Research/armada/blob/master/internal/common/armadaerrors/errors.go#L251

I think the plan was to move to using the appropriate error types rather than setting codes explicitly